android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiRecyclerViewHolder.java

```diff
@@ -139,25 +182,26 @@ protected Drawable generateRippleDrawable(Drawable drawable, String color)
 	 */
 	protected Drawable generateSelectedDrawable(KrollDict properties, Drawable drawable)
 	{
-		if (properties.containsKeyAndNotNull(TiC.PROPERTY_SELECTED_BACKGROUND_COLOR)
-			|| properties.containsKeyAndNotNull(TiC.PROPERTY_SELECTED_BACKGROUND_IMAGE)) {
+		final StateListDrawable stateDrawable = new StateListDrawable();
+
+		if (properties.containsKeyAndNotNull(TiC.PROPERTY_BACKGROUND_SELECTED_COLOR)
+			|| properties.containsKeyAndNotNull(TiC.PROPERTY_BACKGROUND_SELECTED_IMAGE)) {
 
-			final StateListDrawable stateDrawable = new StateListDrawable();
 			final Drawable selectedBackgroundDrawable = TiUIHelper.buildBackgroundDrawable(
-				properties.getString(TiC.PROPERTY_SELECTED_BACKGROUND_COLOR),
-				properties.getString(TiC.PROPERTY_SELECTED_BACKGROUND_IMAGE),
+				properties.getString(TiC.PROPERTY_BACKGROUND_SELECTED_COLOR),
+				properties.getString(TiC.PROPERTY_BACKGROUND_SELECTED_IMAGE),
 				TiConvert.toBoolean(properties.get(TiC.PROPERTY_BACKGROUND_REPEAT), false),
 				null
 			);
 
-			stateDrawable.addState(
-				new int[] { android.R.attr.state_activated }, selectedBackgroundDrawable);
-			stateDrawable.addState(new int[] {}, drawable);
-
-			return stateDrawable;
+			stateDrawable.addState(new int[] { android.R.attr.state_activated }, selectedBackgroundDrawable);
+		} else {
+			stateDrawable.addState(new int[] { android.R.attr.state_activated }, new ColorDrawable(COLOR_SELECTED));
 		}
 
-		return drawable;
+		stateDrawable.addState(new int[] {}, drawable);
```

Comment:
Calling `stateDrawable.addState()` causes an exception to be thrown on Android 6.0 and older.
```
[ERROR] TiExceptionHandler: (main) [198,325] Attempt to invoke virtual method 'android.graphics.drawable.shapes.Shape android.graphics.drawable.shapes.Shape.clone()' on a null object reference
[ERROR] TiExceptionHandler:
[ERROR] TiExceptionHandler:     android.graphics.drawable.ShapeDrawable.mutate(ShapeDrawable.java:501)
[ERROR] TiExceptionHandler:     android.graphics.drawable.LayerDrawable.mutate(LayerDrawable.java:936)
[ERROR] TiExceptionHandler:     android.graphics.drawable.RippleDrawable.mutate(RippleDrawable.java:891)
[ERROR] TiExceptionHandler:     android.graphics.drawable.DrawableContainer.selectDrawable(DrawableContainer.java:443)
[ERROR] TiExceptionHandler:     android.graphics.drawable.StateListDrawable.onStateChange(StateListDrawable.java:111)
[ERROR] TiExceptionHandler:     android.graphics.drawable.StateListDrawable.addState(StateListDrawable.java:94)
[ERROR] TiExceptionHandler:     ti.modules.titanium.ui.widget.listview.TiRecyclerViewHolder.generateSelectedDrawable(TiRecyclerViewHolder.java:202)
```

Reply:
It's easy to reproduce. You just have to create a simple ListView and the default drawable the above method creates will cause this exception to be thrown on older Android OS versions.
```javascript
function createListViewSectionItems() {
	const items = [];
	for (let index = 1; index <= 500; index++) {
		items.push({ properties: { title: "Row " + index } });
	}
	return items;
}
const window = Ti.UI.createWindow();
window.add(Ti.UI.createListView({
	sections: [Ti.UI.createListSection({
		headerTitle: "ListView",
		items: createListViewSectionItems(),
	})],
}));
window.open();
```

Reply:
@garymathews , I'm still seeing a crash on this line. It happens with TableView on Android 5.0 when I set a row's "backgroundColor" property.
```javascript
const defaultRowColor = OS_IOS ? "white" : "?attr/colorBackground";
const rows = [];
for (let index = 1; index <= 20; index++) {
	rows.push(Ti.UI.createTableViewRow({
		title: "Row " + index,
		backgroundColor: defaultRowColor,
	}));
}
const window = Ti.UI.createWindow({ extendSafeArea: false });
const tableView = Ti.UI.createTableView({ data: rows });
tableView.addEventListener("click", (e) => {
	if (e.row.backgroundColor === "orange") {
		e.row.backgroundColor = defaultRowColor;
	} else {
		e.row.backgroundColor = "orange";
	}
});
window.add(tableView);
window.open();
```